hooks/tk-houdini_scene_operations.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+
+from tank import Hook
+
+import hou
+
+class BreakdownSceneOperations(Hook):
+    """
+    Breakdown operations for Houdini.
+
+    This implementation handles detection of alembic node paths.
+    """
+
+    def scan_scene(self):
+        """
+        The scan scene method is executed once at startup and its purpose is
+        to analyze the current scene and return a list of references that are
+        to be potentially operated on.
+
+        The return data structure is a list of dictionaries. Each scene reference
+        that is returned should be represented by a dictionary with three keys:
+
+        - "node": The name of the 'node' that is to be operated on. Most DCCs have
+          a concept of a node, path or some other way to address a particular
+          object in the scene.
+        - "type": The object type that this is. This is later passed to the
+          update method so that it knows how to handle the object.
+        - "path": Path on disk to the referenced object.
+
+        Toolkit will scan the list of items, see if any of the objects matches
+        any templates and try to determine if there is a more recent version
+        available. Any such versions are then displayed in the UI as out of date.
+        """
+
+        # get a list of alembic nodes in the file
+        alembic_nodes = hou.nodeType(hou.sopNodeTypeCategory(),
+            "alembic").instances()
```

Comment:
This is scanning for regular houdini alembic sops (essentially read nodes). The tk-houdini-alembicnode and tk-houdini-mantranode are rops so there's no input path to update for them.